core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -125,8 +126,10 @@ class NettyBlockRpcServer(
         if (blockStored) {
           responseContext.onSuccess(ByteBuffer.allocate(0))
         } else {
-          val exception = new Exception(s"Upload block for $blockId failed. This mostly happens " +
-            s"when there is not sufficient space available to store the block.")
+          val exception = SparkException.internalError(
+            s"Upload block for $blockId failed. This mostly happens " +
+            s"when there is not sufficient space available to store the block.",
```

Comment:
s is not needed here
```suggestion
            "when there is not sufficient space available to store the block.",
```